### PR TITLE
ci: Deploy demo to appspot nightly and on releases

### PR DESCRIPTION
**Pull request by joeyparrish**
This imports our appspot code into the repo, updates all of it to
Python 3, and adds GitHub Actions for deployment.

### CODE REVIEW

File: .github/workflows/custom-actions/prep-for-appspot/action.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - name: Backup demo node modules
+      shell: bash
+      run: |
+        DEMO_NODE_MODULES=$(jq -r .shakaCustom.demoDeps[] package.json)
+        tar cf demo-modules.tar \
+            $(echo "$DEMO_NODE_MODULES" | sed -e 's@^@node_modules/@')
+
+    - name: Delete unneeded files
+      shell: bash
+      run: rm -rf .git .github build test node_modules
```
Comment (theodab) [inline .github/workflows/custom-actions/prep-for-appspot/action.yaml:44]: Wouldn't it be safe to delete the externs folder too?

File: app-engine/demo-version-index/main.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+from flask import Flask, render_template
+
+DEMO_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/'
+
+# In Google Hosted Libraries
+HOSTED_URL_TEMPLATE = 'https://ajax.googleapis.com/ajax/libs/shaka-player/{0}/shaka-player.{1}'
+
+# Before Google Hosted Libraries, v1 appspot URLs
+V1_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/shaka-player.{1}'
+# Before Google Hosted Libraries, v2 and v3 appspot URLs
```
Comment (theodab) [inline app-engine/demo-version-index/main.py:23]: I feel like this should say "v2 and above appspot URLs"
Unless we are planning on changing the folder structure for v4 and I forgot about it, I guess?

File: app-engine/demo-version-index/main.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+V2_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/dist/shaka-player.{1}'
+
+def version_to_demo_url(v):
+  return DEMO_URL_TEMPLATE.format(v.replace('.', '-'))
+
+def version_to_lib_url(v):
+  if v == 'nightly':
+    return V2_URL_TEMPLATE.format(v.replace('.', '-'), 'compiled.js')
```
Comment (theodab) [inline app-engine/demo-version-index/main.py:31]: If v == 'nightly', do we really need to replace all of the dots with dashes?

**Review by theodab — CHANGES_REQUESTED** [review]

File: app-engine/demo-version-index/main.py (hunk 3 above)
Comment (joeyparrish) [reply app-engine/demo-version-index/main.py:31]: No, not at all.  Fixed.

File: .github/workflows/custom-actions/prep-for-appspot/action.yaml (hunk 1 above)
Comment (joeyparrish) [reply .github/workflows/custom-actions/prep-for-appspot/action.yaml:44]: True.  Didn't occur to me before.  Done!

**Review by theodab — APPROVED** [review]

**Pull request merged**